src/styles/main.scss

```diff
@@ -1,3 +1,56 @@
+$days-count: 31;
+$day-block-size: 100px;
+$days-name: mon 0, tue 1, wed 2, thu 3, fri 4, sat 5, sun 6;
+
 body {
   margin: 0;
+  font-family: Arial, Helvetica, sans-serif;
+}
+
+.calendar {
+  margin: 0;
+  padding: 0 10px;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width:  7 * $day-block-size + 20px + 6px;
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    box-sizing: border-box;
+    border: 1px solid black;
+    background-color: #eee;
+    width: $day-block-size;
+    height: $day-block-size;
+    cursor: pointer;
+
+    transition: 500ms;
+  }
+
+  &__day:hover {
+    transform: translate(0,-20px);
+    background-color: pink;
+  }
+
+  @for $day__num from 1 through $days-count {
+    &__day:nth-child(3n + #{$day__num})::before {
+      content: "#{$day__num}";
+      font-size:  30px;
+    }
+  }
+
+  @each $day, $ind in $days-name {
+    &--start-day-#{$day} &__day:first-child {
+      margin-left: $ind * ($day_block_size + 1px);
+    }
+  }
+
+  @for $last-day from 28 through 31 {
+    &--month-length-#{$last-day} &__day:nth-child(n + #{$last-day + 1}) {
+      display: none;
+    }
+  }
 }
```

Comment:
good practice is to leave 1 line at the end of the file (please do this fix in each your file)